plugin/storage/es/mappings/mapping_test.go

```diff
@@ -0,0 +1,316 @@
+				useILM:      true,
+			},
+			mockNewTextTemplateBuilder: func() es.TemplateBuilder {
+				tb := mocks.TemplateBuilder{}
+				ta := mocks.TemplateApplier{}
+				ta.On("Execute", mock.Anything, mock.Anything).Return(nil)
+				tb.On("Parse", mock.Anything).Return(&ta, nil)
+				return &tb
+			},
+			err: "",
+		},
+		{
+			name: "ES Version < 7 Service Error",
+			args: args{
+				shards:      3,
+				replicas:    3,
+				esVersion:   6,
+				indexPrefix: "test",
+				useILM:      true,
+			},
+			mockNewTextTemplateBuilder: func() es.TemplateBuilder {
+				tb := mocks.TemplateBuilder{}
+				ta := mocks.TemplateApplier{}
+				ta.On("Execute", mock.Anything, mock.Anything).Return(nil).Once()
+				ta.On("Execute", mock.Anything, mock.Anything).Return(errors.New("template load error")).Once()
+				tb.On("Parse", mock.Anything).Return(&ta, nil)
+				return &tb
+			},
+			err: "template load error",
+		},
+		{
+			name: "ES Version < 7 Span Error",
+			args: args{
+				shards:      3,
+				replicas:    3,
+				esVersion:   6,
+				indexPrefix: "test",
+				useILM:      true,
+			},
+			mockNewTextTemplateBuilder: func() es.TemplateBuilder {
+				tb := mocks.TemplateBuilder{}
+				ta := mocks.TemplateApplier{}
+				ta.On("Execute", mock.Anything, mock.Anything).Return(errors.New("template load error"))
+				tb.On("Parse", mock.Anything).Return(&ta, nil)
+				return &tb
+			},
+			err: "template load error",
+		},
+		{
+			name: "ES Version  7 Span Error",
+			args: args{
+				shards:      3,
+				replicas:    3,
+				esVersion:   7,
+				indexPrefix: "test",
+				useILM:      true,
+			},
+			mockNewTextTemplateBuilder: func() es.TemplateBuilder {
+				tb := mocks.TemplateBuilder{}
+				ta := mocks.TemplateApplier{}
+				ta.On("Execute", mock.Anything, mock.Anything).Return(errors.New("template load error")).Once()
+				tb.On("Parse", mock.Anything).Return(&ta, nil)
+				return &tb
+			},
+			err: "template load error",
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			mappingBuilder := MappingBuilder{
+				TemplateBuilder: test.mockNewTextTemplateBuilder(),
+				Shards:          test.args.shards,
+				Replicas:        test.args.replicas,
+				EsVersion:       test.args.esVersion,
+				IndexPrefix:     test.args.indexPrefix,
+				UseILM:          test.args.useILM,
+			}
+			_, _, err := mappingBuilder.GetSpanServiceMappings()
+			if test.err != "" {
+				assert.EqualError(t, err, test.err)
+			} else {
+				assert.NoError(t, err)
+			}
+		})
+	}
+}
+
+func TestMappingBuilder_GetDependenciesMappings(t *testing.T) {
+	tb := mocks.TemplateBuilder{}
+	ta := mocks.TemplateApplier{}
+	ta.On("Execute", mock.Anything, mock.Anything).Return(errors.New("template load error"))
+	tb.On("Parse", mock.Anything).Return(&ta, nil)
+
+	mappingBuilder := MappingBuilder{
+		TemplateBuilder: &tb,
+		Shards:          5,
+		Replicas:        5,
+		EsVersion:       7,
+		IndexPrefix:     "",
+		UseILM:          false,
```

Comment:
these could be removed since the test isn't asserting on the resulting mapping string

Reply:
Yep, Removed in next commit